lib/executor.js

```diff
@@ -330,24 +334,42 @@ function setupMiddleware(app, instructions) {
     }
     assert(typeof factory === 'function',
       'Middleware factory must be a function');
-    data.config = getUpdatedConfigObject(app, data.config);
+    var opts = {
+      useEnvVars: true,
+    };
+    data.config = getUpdatedConfigObject(app, data.config, opts);
     app.middlewareFromConfig(factory, data.config);
   });
 }
 
-function getUpdatedConfigObject(app, config) {
+function getUpdatedConfigObject(app, config, opts) {
   var DYNAMIC_CONFIG_PARAM = /\$\{(\w+)\}$/;
+  var useEnvVars = opts.useEnvVars || undefined;
```

Comment:
Are you sure `opts` is always set? I'd prefer to make it an option argument and check that it's set. Also `|| undefined` is not needed, it just converts falsy value like `0`, `null`, empty string or `false` to `undefined`, which is also falsy.

```js
var useEnvVars = opts && opts.useEnvVars;
```